gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/reporter/EventReporter.java

```diff
@@ -189,13 +212,20 @@ public void run() {
     protected MetricFilter filter;
     protected TimeUnit rateUnit;
     protected TimeUnit durationUnit;
+    protected Config config;
 
     protected Builder(MetricContext context) {
       this.context = context;
       this.name = "MetricReportReporter";
       this.rateUnit = TimeUnit.SECONDS;
       this.durationUnit = TimeUnit.MILLISECONDS;
       this.filter = MetricFilter.ALL;
+      this.config = ConfigFactory.empty();
+    }
+
+    public T withConfig(Config config) {
```

Comment:
Should we be more explicit and have methods withQueueCapacity(int) and withReportingInterval(int)?